starlette/formparsers.py

```diff
@@ -38,6 +38,12 @@ def _user_safe_decode(src: bytes, codec: str) -> str:
         return src.decode("latin-1")
 
 
+class MultiPartException(Exception):
+    def __init__(self, message: str, *args: typing.Any) -> None:
```

Comment:
Why do we need to define `__init__` here? Are we using `*args`?

Reply:
No, only the message. I've removed when you mentioned this, but I need to put it back. Conceptually, `MultiPartException` is not an `HTTPException`, so it needs to inherit from `Exception`. The message is necessary here because there will be another part of the code that will raise this error in a future PR (the "name" mentioned on the description).

Reply:
I still don't understand why it needs to be here, standard behavior for exceptions is that the first argument is a string which corresponds to the exception message. Is this just for when `"app"` is not in scope? If so users can just parse the exceptions, I don't think it's a use case we should be putting effort into supporting.

Reply:
Let's continue on the other message.